packages/rest-client/src/services/promise/updateOne/updateOne.test.ts

```diff
@@ -0,0 +1,53 @@
+import { describe, it, expect } from "vitest"
+import { createStore, convertResourceToRecord } from "../../store"
+import type { Schema, Source } from "../../types"
+import { updateOne } from "./updateOne"
+
+const fakeDataSource: Source = {
+  version: 0,
+  getList: () => Promise.resolve([]),
+  getOne: () => Promise.resolve([]),
+  createOne: () => Promise.resolve([]),
+  updateOne: () =>
+    Promise.resolve([
+      {
+        id: "1",
+        __schema: "Article",
+        attributes: { title: "updated title", body: "baz-body" },
+      },
+    ]),
+}
+
+const ArticleSchema = { name: "Article" } as Schema
+
+describe("rest-client/services/promise/updateOne", () => {
+  const data = {
+    id: "1",
+    attributes: { title: "updated title" },
+  }
+  const expected = {
+    id: "1",
+    __schema: "Article",
+    attributes: { title: "updated title", body: "baz-body" },
+  }
+
+  it("should return the new record", async () => {
+    createStore(["Article"])
+    const result = await updateOne(fakeDataSource, ArticleSchema, data)
+    expect(result).toEqual(convertResourceToRecord(expected))
+  })
+
+  it.todo("should insert the record into the store")
```

Comment:
these tests are just marked as todo because the store stuff is going on pause for a bit, in a (near) future PR I will make changes to always refetch data on subscribe changes rather than use store